### PR TITLE
Update `group_type` enum

### LINKED ISSUE #77: Update specification for `group_types`
We should update the docs to include

- `apartment`,
- `house`,
- `home`, 
- `trip`,
- `couple`,
- and `other`

### PR DESCRIPTION
**Pull request by dylanrainwater**
Closes #77. Updates references to `group_type` to include `home` and `couple`. Also includes notes and updates references to prefer `home` to `apartment` or `house`.

### CODE REVIEW

File: paths/create_group.yaml [hunk 1]
```diff
@@ -19,8 +19,12 @@ post:
               type: string
             group_type:
               type: string
-              description: What is the group used for?
-              enum: [apartment, house, trip, other]
+              enum: [home, trip, couple, other, apartment, house]
```
Comment (dylanrainwater) [inline paths/create_group.yaml:22]: - Moved `enum` up.
- Added `home` and `couple`
- Moved `apartment` and `house` to end of list

File: paths/create_group.yaml [hunk 2]
```diff
@@ -19,8 +19,12 @@ post:
...
+              example: "home"
```
Comment (dylanrainwater) [inline paths/create_group.yaml:23]: - Added example

File: paths/create_group.yaml [hunk 3]
```diff
@@ -19,8 +19,12 @@ post:
...
+              description: |
+                What is the group used for?
+
+                **Note**: It is recommended to use `home` in place of `house` or `apartment`.
```
Comment (dylanrainwater) [inline paths/create_group.yaml:27]: - Added Note to description

File: schemas/group.yaml [hunk 4]
```diff
@@ -8,8 +8,12 @@ properties:
     example: "Housemates 2020"
   group_type:
     type: string
-    enum: [apartment, house, trip, other]
-    example: "apartment"
+    enum: [home, trip, couple, other, apartment, house]
```
Comment (dylanrainwater) [inline schemas/group.yaml:11]: - Added `home` and `couple`
- Moved `apartment` and `house` to end of list

File: schemas/group.yaml [hunk 5]
```diff
@@ -8,8 +8,12 @@ properties:
...
+    example: "home"
```
Comment (dylanrainwater) [inline schemas/group.yaml:12]: - Changed example from `apartment` to `home`

File: schemas/group.yaml [hunk 6]
```diff
@@ -8,8 +8,12 @@ properties:
...
+    description: |
+      What is the group used for?
+
+      **Note**: It is recommended to use `home` in place of `house` or `apartment`.
```
Comment (dylanrainwater) [inline schemas/group.yaml:16]: - Added a description (copied from other example)

File: splitwise.yaml [hunk 7]
```diff
@@ -277,7 +277,7 @@ tags:
   - name: groups
     x-displayName: Groups
     description:
-      A Group represents a collection of users who share expenses together. For example, some users use a Group to aggregate expenses related to an apartment. Others use it to represent a trip.
+      A Group represents a collection of users who share expenses together. For example, some users use a Group to aggregate expenses related to a home. Others use it to represent a trip.
```
Comment (dylanrainwater) [inline splitwise.yaml:280]: - Updated reference of `apartment` to `home`

**Review by dylanrainwater — COMMENTED** [review]
some notes:

**Review by jas14 — APPROVED** [review]
:shipit:

**Pull request merged**